### PR TITLE
fix(sdk): exp SDK tests and fix

### PR DESCRIPTION
**Pull request by pflynn-virtru**
This pull request adds comprehensive regression and compatibility tests to the experimental TDF SDK, focusing on key management edge cases and ensuring encryption format compatibility with the production SDK. The new tests verify correct handling of legacy attribute grants, segment integrity hashing, and cross-decryption between SDKs. Additionally, new imports were added to support these tests.

See xtest failure: https://github.com/opentdf/tests/pull/414

**Regression and Compatibility Testing Enhancements:**

* Added regression tests to `xor_splitter_test.go` to ensure that when an attribute grant references a KAS URL by URI only (without an embedded public key), the default KAS public key is correctly merged, and that existing keys are not overwritten by the default.
* Added end-to-end regression tests in `writer_test.go` to verify that segment hashes include both nonce and ciphertext (not just ciphertext), and that finalization succeeds for URI-only grants by leveraging the default KAS public key.

**Cross-SDK Compatibility:**

* Introduced a new test in `writer_test.go` to validate that the experimental writer's output is fully compatible with the production SDK's decryption and segment hash verification, including full TDF ZIP assembly and parsing with the production ZIP reader.

**Test Infrastructure Improvements:**

* Registered the new regression and compatibility tests in the main test suite.
* Added necessary imports for cryptographic operations and ZIP parsing to support the new tests. [[1]](diffhunk://#diff-f935252b0fdbd2ed0baff420a925250cdcfef0dd3c877aa984d449b96d93340aR7-R12) [[2]](diffhunk://#diff-f935252b0fdbd2ed0baff420a925250cdcfef0dd3c877aa984d449b96d93340aR21)

### CODE REVIEW

**Review by dmihalcik-virtru — DISMISSED** [review]

**Review by dmihalcik-virtru — DISMISSED** [review]

File: examples/cmd/benchmark_experimental.go [hunk 1]
```diff
@@ -98,12 +109,48 @@ func runExperimentalWriterBenchmark(_ *cobra.Command, _ []string) error {
...
+	}
+	var decrypted bytes.Buffer
+	if _, err = io.Copy(&decrypted, tdfReader); err != nil {
+		return fmt.Errorf("failed to decrypt TDF with production SDK: %w", err)
+	}
+
+	if bytes.Equal(payload, decrypted.Bytes()) {
+		fmt.Println("| Decrypt Verify     | PASS - roundtrip matches |")
+	} else {
+		fmt.Printf("| Decrypt Verify     | FAIL - payload %d bytes, decrypted %d bytes |\n", len(payload), decrypted.Len())
```
Comment (dmihalcik-virtru) [inline examples/cmd/benchmark_experimental.go:152]: Is this accessible? I'd think `Copy` would return an error in most cases I can think of, unless maybe somehow you opened the wrong file?

File: examples/cmd/benchmark_experimental.go [hunk 2]
```diff
@@ -98,12 +109,48 @@ func runExperimentalWriterBenchmark(_ *cobra.Command, _ []string) error {
...
 	totalTime := end.Sub(start)
 
+	// Assemble the complete TDF: segment data (in order) + finalize data
+	var tdfBuf bytes.Buffer
+	for i, sr := range segResults {
+		if _, err := io.Copy(&tdfBuf, sr.TDFData); err != nil {
+			return fmt.Errorf("failed to read segment %d TDF data: %w", i, err)
+		}
+	}
+	tdfBuf.Write(result.Data)
```
Comment (dmihalcik-virtru) [inline examples/cmd/benchmark_experimental.go:119]: Why not just copy to `io.Discard` and record the number of bytes? There isn't much advantage to storing the file twice in memory. See the note below: decrypt is authenticated so mismatched bytes should be unreachable

**Review by dmihalcik-virtru — APPROVED** [review]
